model/manifest.go

```diff
@@ -304,6 +317,109 @@ func (m *Manifest) MeetMinServerVersion(serverVersion string) (bool, error) {
 	return true, nil
 }
 
+func (m *Manifest) IsValid() error {
+	if !IsValidPluginId(m.Id) {
+		return errors.New("invalid plugin ID")
+	}
+
+	if m.HomepageURL == "" || !IsValidHttpUrl(m.HomepageURL) {
+		return errors.New("invalid HomepageURL")
+	}
+
+	if m.SupportURL == "" || !IsValidHttpUrl(m.SupportURL) {
+		return errors.New("invalid SupportURL")
+	}
+
+	_, err := semver.Parse(m.Version)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse Version")
+	}
+
+	_, err = semver.Parse(m.MinServerVersion)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse MinServerVersion")
+	}
+
+	err = isValidSettingsSchema(m.SettingsSchema)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func isValidSettingsSchema(settingsSchema *PluginSettingsSchema) error {
```

Comment:
What do you think about making this also a method?
```suggestion
func (s *settingsSchema) isValidS() error {
```

Missed this in the first review